src/lib/client.js

```diff
@@ -3,23 +3,27 @@
 const assert = require('assert')
 const WebSocket = require('ws')
 const debug = require('debug')('ilp-plugin-payment-channel-framework:client')
-const url = require('url')
+const { URL } = require('url')
 
 module.exports = class BtpClient {
-  constructor ({ server, secret, plugin, insecure }) {
-    this._server = server
+  constructor ({ btpUri, plugin }) {
     this._plugin = plugin
 
-    // The server URI must follow the format: btp+wss://host:port/path
-    // See also: https://github.com/interledger/interledger/wiki/Interledger-over-BTP
-    const parsedServer = url.parse(server)
-    assert(parsedServer.protocol.startsWith('btp+'), 'server uri must start with "btp+"')
-    this._wsUri = (insecure ? 'ws://' : 'wss://') + parsedServer.host + parsedServer.path
```

Comment:
I'm unclear on what's wrong with this way of constructing the URL. Using a substring seems like a jankier solution. Host already contains the port, and I think it's a good idea to require a flag in order to allow an insecure connection to the server. We may want to throw an error if the protocol is `btp+ws` and `insecure` is `false`, though.

Reply:
> require a flag in order to allow an insecure connection to the server

ok. Constructing the URL from parts if overwriting functionality from the URL class, so that's not a good idea, I think. What might be an option would be to fix it upstream so that we can do:
```js
> const { URL } = require('url')
undefined
> a = new URL('btp+wss://user:pass@example.com/asdf')
URL {
  href: btp+wss://user:pass@example.com/asdf
  protocol: btp+wss:
  username: user
  password: --------
  hostname: example.com
  pathname: /asdf
}
> a.toString()
'btp+wss://user:pass@example.com/asdf'
> a.protocol = 'wss:'
'wss:'
> a.toString()
'btp+wss://user:pass@example.com/asdf'
>
```